tests/trainer/test_config_validator.py

```diff
@@ -92,3 +92,23 @@ def test_test_loop_config(tmpdir):
         model = EvalModelTemplate(**hparams)
         model.test_step = None
         trainer.test(model, test_dataloaders=model.dataloader(train=False))
+
+
+def test_validation_loop_config(tmpdir):
+    """"
+    When either validation loop or validation data are missing
+    """
+    hparams = EvalModelTemplate.get_default_hparams()
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
+
+    # has val loop but no val data
+    with pytest.warns(UserWarning):
+        model = EvalModelTemplate(**hparams)
```

Comment:
You dont need to pass `**hparams` if you are not modifying them

Reply:
All tests in the file are like that, should I update them all?

Reply:
IMO, considering that this is not a small PR, you dont want to add the extra overhead for reviewers by updating many unrelated tests.

I would leave them as they are and maybe update them in a separate PR.

Reply:
Alright, thanks!